project/MimaExcludes.scala

```diff
@@ -70,6 +70,13 @@ object MimaExcludes {
           "org.apache.spark.scheduler.AskPermissionToCommitOutput.this"),
         ProblemFilters.exclude[IncompatibleMethTypeProblem](
           "org.apache.spark.scheduler.AskPermissionToCommitOutput.apply")
+      ) ++ Seq(
+        // SPARK-10704: Switched ShuffleManager from a trait to an abstract class.
```

Comment:
This will break client code which called these APIs directly, but these were clearly labeled as `private[spark]` and shouldn't have been used in this way.